src/docfx/config/ops/OpsConfigAdapter.cs

```diff
@@ -227,24 +240,56 @@ private static string GetHostName(string siteName)
             switch (siteName)
             {
                 case "DocsAzureCN":
-                    return s_isProduction ? "docs.azure.cn" : "ppe.docs.azure.cn";
+                    return s_docsEnvironment switch
+                    {
+                        DocsEnvironment.Prod => "docs.azure.cn",
+                        DocsEnvironment.PPE => "ppe.docs.azure.cn",
+                        DocsEnvironment.Internal => "ppe.docs.azure.cn",
+                        DocsEnvironment.Perf => "ppe.docs.azure.cn",
+                        _ => throw new NotSupportedException()
+                    };
                 case "dev.microsoft.com":
-                    return s_isProduction ? "developer.microsoft.com" : "devmsft-sandbox.azurewebsites.net";
+                    return s_docsEnvironment switch
+                    {
+                        DocsEnvironment.Prod => "developer.microsoft.com",
+                        DocsEnvironment.PPE => "devmsft-sandbox.azurewebsites.net",
+                        DocsEnvironment.Internal => "devmsft-sandbox.azurewebsites.net",
+                        DocsEnvironment.Perf => "devmsft-sandbox.azurewebsites.net",
+                        _ => throw new NotSupportedException()
+                    };
                 case "rd.microsoft.com":
-                    return "rd.microsoft.com";
+                    return s_docsEnvironment switch
+                    {
+                        DocsEnvironment.Prod => "rd.microsoft.com",
+                        _ => throw new NotSupportedException()
+                    };
                 default:
-                    return s_isProduction ? "docs.microsoft.com" : "ppe.docs.microsoft.com";
+                    return s_docsEnvironment switch
+                    {
+                        DocsEnvironment.Prod => "docs.microsoft.com",
+                        DocsEnvironment.PPE => "ppe.docs.microsoft.com",
+                        DocsEnvironment.Internal => "ppe.docs.microsoft.com",
+                        DocsEnvironment.Perf => "ppe.docs.microsoft.com",
+                        _ => throw new NotSupportedException()
+                    };
             }
         }
 
         private static string GetXrefHostName(string siteName, string branch)
         {
-            return !IsLive(branch) && s_isProduction ? $"review.{GetHostName(siteName)}" : GetHostName(siteName);
+            return !IsLive(branch) && s_docsEnvironment == DocsEnvironment.Prod ? $"review.{GetHostName(siteName)}" : GetHostName(siteName);
         }
 
         private static bool IsLive(string branch)
         {
             return branch == "live" || branch == "live-sxs";
         }
+
+        private static DocsEnvironment GetDocsEnvironment()
+        {
+            return Enum.TryParse(Environment.GetEnvironmentVariable("DOCS_ENVIRONMENT"), true, out DocsEnvironment docsEnvironment)
+                ? docsEnvironment
+                : DocsEnvironment.PPE;
```

Comment:
When `DOCS_ENVIRONMENT` is not set, fallback to `Prod` environment. Because for docfx shipped to an end user, it should connect to our production environment without user doing anything special.

```suggestion
                : DocsEnvironment.Prod;
```